### PR TITLE
add a processingElement hardware-in-the-loop test

### PR DESCRIPTION
**Pull request by cuddlefishie**
This adds a `bittide-instance` that tests that a RISC-V core can run a program in a HITL-setup.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -689,6 +721,34 @@ jobs:
...
+        run: |
+          find . -type f \
+               -not -path './.git/*' \
+            -a -not -path './firmware/examples/target/*' \
+            -a -not -path './clash-vexriscv/clash-vexriscv/build_out_dir/*' \
+            -a -not -path './clash-vexriscv/clash-vexriscv/example-cpu/target/*' \
+            -a -not -path './clash-vexriscv/clash-vexriscv/example-cpu/project/target/*' \
+            -a -not -path './_build/*' \
+            -a -not -path './target/*' \
+            -a -not -path './dist-newstyle/*' \
```
Comment (martijnbastiaan) [inline .github/workflows/ci.yml:750]:
```suggestion
      - name: Show project files
        run: |
          git ls-files
```

? Or is this not what you're after?

File: bittide-instances/src/Bittide/Instances/Tests/VexRiscv.hs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                    1 -> Success
+                    _ -> Fail
+              in (state, (emptyWishboneS2M { acknowledge = True }, state))
+
+    {-
+    0b1000xxxx 0x8x instruction memory
+    0b0100xxxx 0x4x data memory
+    0b1100xxxx 0xCx status register
+    -}
+    peConfig = PeConfig (0b10 :> 0b01 :> 0b11 :> Nil) (Reloadable $ Blob iMem) (Reloadable $ Blob dMem)
```
Comment (martijnbastiaan) [inline bittide-instances/src/Bittide/Instances/Tests/VexRiscv.hs:81]: Maybe add a comment that we want `NonReloadable` for this, but it breaks synthesis. Could you create an issue?

File: bittide/src/Bittide/ProcessingElement/Util.hs [hunk 3]
```diff
@@ -66,8 +66,8 @@ getBytesMems elfPath maybeDeviceTree = do
       "Bittide.ProcessingElement.Util: Overlapping element in data memory and device tree at address 0x"
       <> showHex k "") dMem0 deviceTreeMap
 
-  putStrLn $ "elf file: " <> elfPath <>
-          "\ndevice tree: " <> show (fmap (\(a,b) -> (a,b, L.length b)) maybeDeviceTree)
+  -- putStrLn $ "elf file: " <> elfPath <>
+  --         "\ndevice tree: " <> show (fmap (\(a,b) -> (a,b, L.length b)) maybeDeviceTree)
```
Comment (martijnbastiaan) [inline bittide/src/Bittide/ProcessingElement/Util.hs:70]: Just remove?

File: bittide-instances/src/Bittide/Instances/Tests/VexRiscv.hs [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+    unitC = CSignal (pure ())
+    (_, CSignal status) = circuitFn ((), unitC)
+
+    Circuit circuitFn = circuit $ \unit -> do
+        [wb] <- processingElement peConfig -< unit
+        statusRegister -< wb
+
+    statusRegister :: Circuit (Wishbone dom 'Standard 30 (Bytes 4)) (CSignal dom TestStatus)
+    statusRegister = Circuit $ \(fwd, CSignal _) ->
```
Comment (martijnbastiaan) [inline bittide-instances/src/Bittide/Instances/Tests/VexRiscv.hs:52]: @lmbollen Is there a generic component for this?

**Review by martijnbastiaan — APPROVED** [review]
LGTM!

File: bittide-instances/src/Bittide/Instances/Tests/VexRiscv.hs (hunk 4 above)
Comment (cuddlefishie) [reply bittide-instances/src/Bittide/Instances/Tests/VexRiscv.hs:52]: There is `wbToVec` https://github.com/bittide/bittide-hardware/blob/b244b2af2de71ee36b26391670a820f3188a1482/bittide/src/Bittide/Wishbone.hs#L336 in #354 which could be adapted to be used in a circuit like this, but there's no generalised version like that AFAIK

File: bittide-instances/src/Bittide/Instances/Tests/VexRiscv.hs (hunk 2 above)
Comment (cuddlefishie) [reply bittide-instances/src/Bittide/Instances/Tests/VexRiscv.hs]: Done #372

File: .github/workflows/ci.yml (hunk 1 above)
Comment (martijnbastiaan) [reply .github/workflows/ci.yml]: agreed

File: bittide-instances/src/Bittide/Instances/Tests/VexRiscv.hs (hunk 4 above)
Comment (martijnbastiaan) [reply bittide-instances/src/Bittide/Instances/Tests/VexRiscv.hs:52]: cool cool, let's leave it like it is

Comment (lmbollen) [reply bittide-instances/src/Bittide/Instances/Tests/VexRiscv.hs:52]: We could definitely make a circuit version for `wbToVec`. The main reason I implemented it like this is to easily create interfaces with multiple signals. The reason I have not implemented it as `Signal` is to make it obvious that it's a simple combinatorial component.

**Pull request merged**